### ISSUE TITLE
Show a breakdown of which domains are causing you to lose a green mark

### ISSUE THREAD

**Issue by mrchrisadams** [thegreenwebfoundation/lighthouse-plugin-greenhouse#2]
Right now, if you have a site served entirely by green servers, you'll get something like this:

![Screenshot 2019-07-17 at 18 31 01](https://user-images.githubusercontent.com/17906/61481558-d11d6b00-a998-11e9-8909-71c1812e3757.png)

However, if you have some 'grey' servers in your supply chain, you'll end up with something like this (yes, the copy needs work)

![Screenshot 2019-07-18 at 13 56 22](https://user-images.githubusercontent.com/17906/61481621-f4e0b100-a998-11e9-9e6f-d61853c65935.png)

### Showing which domains are losing the 100 mark

If a site uses green power, we have richer information returned by the Green Web API, that would make it easier to act upon the data, rather than manually checking each link, or seeing what's returned by the plugin results.

It looks like this:

```js
{ green: true,
    checked_on: '2019-07-18T11:39:58+02:00',
    url: 'www.triodos.com',
    data: true,
    hostedby: 'Combell Group NV',
    hostedbyid: 198,
    hostedbywebsite: 'www.combell.com',
    partner: '' },
  { green: true,
    checked_on: '2019-07-18T11:39:58+02:00',
    url: 'video.triodos.com',
    data: true,
    hostedby: 'Combell Group NV',
    hostedbyid: 198,
    hostedbywebsite: 'www.combell.com',
    partner: '' },
  { green: false,
    checked_on: '2019-07-18T11:39:58+02:00',
    url: 't-pan.triodos.com',
    data: true },
  { green: false,
    checked_on: '2019-07-18T11:39:58+02:00',
    url: 'p-pan.triodos.com',
    data: true }
```

This would be enough to create some grouped lists to show which were the 'green', and which were the 'grey' domains, a bit like how lighthouse gives richer info here:
![Screenshot 2019-07-18 at 20 47 04](https://user-images.githubusercontent.com/17906/61483517-3a06e200-a99d-11e9-8fbd-dacd289eea73.png)

### Implementing this 

If you look at some of the existing renderers, you'll see how lighthouse has handlers for passing a set of headers like this:

```js

   /** @type {LH.Audit.Details.Opportunity['headings']} */
    const headings = [
      {key: 'url', valueType: 'thumbnail', label: ''},
      {key: 'url', valueType: 'url', label: str_(i18n.UIStrings.columnURL)},
      {key: 'totalBytes', valueType: 'bytes', label: str_(i18n.UIStrings.columnSize)},
      {key: 'wastedBytes', valueType: 'bytes', label: str_(i18n.UIStrings.columnWastedBytes)},
    ];
```

Followed by building an array of making data like so:

```js
 /** @type {Array<{url: string, fromProtocol: boolean, isCrossOrigin: boolean, totalBytes: number, wastedBytes: number}>} */
    const items = [];
    const warnings = [];
    for (const image of images) {
      if (image.failed) {
        warnings.push(`Unable to decode ${URL.getURLDisplayName(image.url)}`);
        continue;
      } else if (/(jpeg|bmp)/.test(image.mimeType) === false) {
        continue;
      }

      let jpegSize = image.jpegSize;
      let fromProtocol = true;

      if (typeof jpegSize === 'undefined') {
        const imageElement = imageElementsByURL.get(image.url);
        if (!imageElement) {
          warnings.push(`Unable to locate resource ${URL.getURLDisplayName(image.url)}`);
          continue;
        }

        jpegSize = UsesOptimizedImages.estimateJPEGSizeFromDimensions(imageElement);
        fromProtocol = false;
      }

      if (image.originalSize < jpegSize + IGNORE_THRESHOLD_IN_BYTES) continue;

      const url = URL.elideDataURI(image.url);
      const isCrossOrigin = !URL.originsMatch(pageURL, image.url);
      const jpegSavings = UsesOptimizedImages.computeSavings({...image, jpegSize});

      items.push({
        url,
        fromProtocol,
        isCrossOrigin,
        totalBytes: image.originalSize,
        wastedBytes: jpegSavings.bytes,
      });
    }
```

These are then combined to provide the data needed for that expandable widget.  See the `uses-optimized-images` audit below:

https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/byte-efficiency/uses-optimized-images.js#L66-L132

Listing `domain`, `hosted by`, and `green/grey` , then showing these grouped by green then grey would be informative and useful. For each url, we could linking back to something like the link we use below:

https://www.thegreenwebfoundation.org/green-web-check/?url=triodos.com

Or a pared-back version on the report page, to show largely the same data, without needing to leave the page.

**Comment by mrchrisadams** [comment]
I'm moving the `hostedby` stuff into a separate issue, so we can get this the new work me and @jasonneylon have been doing, along with an issue to link to a specific page, like so when checking the result for a page:

https://www.thegreenwebfoundation.org/green-web-check/?url=google.com

This would provide users with some concrete steps they can take to switch links from grey, to green.

**Comment by mrchrisadams** [comment]
Resolved with #10
